src/main/java/com/example/tournamentMaker/tournament/result/ResultService.java

```diff
@@ -0,0 +1,168 @@
+                    }
+
+                    game.setHostPoints(footballResultRequest.getHostPoints());
+                    game.setGuestPoints(footballResultRequest.getGuestPoints());
+                    gameRepository.save(game);
+
+                    FootballStatistics hostStatistics = (FootballStatistics) host.getStatistics();
+                    FootballStatistics guestStatistics = (FootballStatistics) guest.getStatistics();
+
+                    MatchResult hostResult = getHostResult(footballResultRequest);
+                    MatchResult guestResult = getGuestResult(hostResult);
+                    addResultOfTheMatchToStatistics(hostResult, hostStatistics, guestStatistics);
+
+                    updateRecentResult(hostResult, hostStatistics.getRecentMatchResults());
+                    updateRecentResult(guestResult, guestStatistics.getRecentMatchResults());
+
+                    updateStatisticsForIndividualPlayers(footballResultRequest, host, guest, hostStatistics, guestStatistics);
+
+                    statisticsRepository.save(hostStatistics);
+                    statisticsRepository.save(guestStatistics);
+                },
+                () -> {
+                    throw new NoSuchElementException(Constans.NO_TOURNAMENT_FOUND);
+                });
+    }
+
+    private void updateStatisticsForIndividualPlayers(FootballResultRequest footballResultRequest, Team host, Team guest, FootballStatistics hostStatistics, FootballStatistics guestStatistics) {
+        updateSpecificStatisticInTeam(footballResultRequest.getHostStatistics().getShirtNumbersWithGoal(),
+                host, hostStatistics.getPlayersIdWithGoal());
+        updateSpecificStatisticInTeam(footballResultRequest.getHostStatistics().getShirtNumbersWithYellowCard(),
+                host, hostStatistics.getPlayersIdWithYellowCard());
+        updateSpecificStatisticInTeam(footballResultRequest.getHostStatistics().getShirtNumbersWithRedCard(),
+                host, hostStatistics.getPlayersIdWithRedCard());
+        updateSpecificStatisticInTeam(footballResultRequest.getHostStatistics().getGetShirtNumbersWithCleanSlate(),
+                host, hostStatistics.getPlayersIdWithCleanSheets());
+
+        updateSpecificStatisticInTeam(footballResultRequest.getGuestStatistics().getShirtNumbersWithGoal(),
+                guest, guestStatistics.getPlayersIdWithGoal());
+        updateSpecificStatisticInTeam(footballResultRequest.getGuestStatistics().getShirtNumbersWithYellowCard(),
+                guest, guestStatistics.getPlayersIdWithYellowCard());
+        updateSpecificStatisticInTeam(footballResultRequest.getGuestStatistics().getShirtNumbersWithRedCard(),
+                guest, guestStatistics.getPlayersIdWithRedCard());
+        updateSpecificStatisticInTeam(footballResultRequest.getGuestStatistics().getGetShirtNumbersWithCleanSlate(),
+                guest, guestStatistics.getPlayersIdWithCleanSheets());
```

Comment:
jakos patrzac na to nie podoba mi sie wcale ta metoda. 
Trzeba cos pomyslec lepszego z ty m